File: mario.c
// This is a script to generate a Mario split pyramid as part of CS50 Problem Set 1
// Zach Sirera
// 5/13/18

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;

    // Prompt the user for a specified height of pyramid
    // Reject height if it is less than 0 or greater than 23
    do
    {
        n = get_int("Height:  ");
    }
    while (n < 0 || n > 23);

    // Print each row, top to bottom, line i at a time.
    for (int i = 0; i < n; i++)
    {
        // Print the spaces on the left side of the pyramid
        for (int k = (n - 1); k > i; k--)
        {
            printf(" ");
        }

        // Print the # chars on the left side of the pyramid
        for (int l = 0; l <= i; l++)
        {
            printf("#");
        }

        // Print the gap
        printf("  ");

        // Print # chars on the right side of the pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // Print to next line
        printf("\n");
    }
}

File: pset6/pset6/mario/mario.py
# This is a program to implement Mario (more) as part of CS50 Problem Set 6
# Zach Sirera
# 8/2/2018 to 8/10/2018

import cs50

# Ask the user for the height of the Mario Pyramid
print("Height: ", end="")
height = cs50.get_int()

# Validate the height input
if height < 0 or height > 23:
    print("Height: ", end="")
    height = cs50.get_int()

# Carry out the printing iteration line by line
for i in range(height):
    # Print the initial spaces
    for j in range(height - i - 1):
        print(" ", end="")
    # Print the # characters for the left pyramid on each row
    for k in range(i + 1):
        print("#", end="")
    # Print the space between each pyramid
    print("  ", end="")
    # print the # characters for the right pyramid on each row
    for l in range(i + 1):
        print("#", end="")
    print("")